### DISCUSSION TITLE
Is osgDB::writeNodeFile() thread safety?

### DISCUSSION THREAD

**Discussion by husterlzj** [openscenegraph/OpenSceneGraph#1276] [Q&A]
When I use osgDB::writeNodeFile() in mutil-threading, there may be occasional infinite loops, the process cannot exit.

**Comment by robertosfield** [comment]
You'll need to provide more information about what you are doing to be able to guess at what might be going wrong.  What type of data what formats, what's the stack trace of the threads when it gets locked out.

**Reply by husterlzj** [reply]
Thanks for your reply. The type of data is PagedLod node. The data format is .osgb file.
You can refer to the following code：
//
#pragma omp parallel for
for(int i=0;  i<nodeNum;  ++i){
    …
    ExportNodeFile(i)
    {
        …
         osg::ref_ptr<osgDB::ReaderWriter::Options> option = new osgDB:: ReaderWriter::Options;
        option->setOptionString("WriteImageHint=Includefile  JPEG_QUALITY 100"); 
        //  type of node is: osg::PagedLOD, osg::Geode, or osg::Group
        osgDB::writeNodeFile(node, savePath, Option);
        …
    }
}
//
no threads is running when it gets locked out.

**Comment by robertosfield** [comment]
Given the scant detail my best guess is libjpeg functions aren't thread safe. Try getting a stack trace.

**Reply by husterlzj** [reply]
Thank you very much. I'Il try.
